types/index.d.ts

```diff
@@ -0,0 +1,340 @@
+
+    /**
+     * Defaults to `false`
+     *
+     * Locks joystick’s movement to the y (vertical) axis
+     */
+    lockY?: boolean;
+}
+
+export interface Position {
+    x: number;
+    y: number;
+}
+
+export interface JoystickOutputData {
+    angle: {
+        degree: number;
+        radian: number;
+    };
+    direction: {
+        angle: string;
+        x: string;
+        y: string;
+    };
+    distance: number;
+    force: number;
+    identifier: number;
+    instance: Joystick;
+    position: Position;
+    pressure: number;
+}
+
+export type JoystickEventTypes =
+    /**
+     * A joystick is activated. (the user pressed on the active zone)
+     * Will pass the instance alongside the event.
+     */
+    | 'start'
+
+    /** A joystick is de-activated. (the user released the active zone)
+     * Will pass the instance alongside the event.
+     */
+    | 'end'
+
+    /**
+     * A joystick is moved
+     */
+    | 'move'
+
+    /**
+     * When a direction is reached after the threshold.
+     *
+     * Direction are split with a 45° angle.
+     */
+    | 'dir'
+    | 'dir:up'
+    | 'dir:down'
+    | 'dir:right'
+    | 'dir:left'
+
+    /**
+     * When a plain direction is reached after the threshold.
+     * Plain directions are split with a 90° angle.
+     */
+    | 'plain'
+
+    // plain variations
+    | 'plain:up'
+    | 'plain:down'
+    | 'plain:right'
+    | 'plain:left'
+
+    /**
+     * Is triggered at the end of the fade-in animation.
+     * Will pass the instance alongside the event.
+     *
+     * Won’t be trigger in a dataOnly configuration.
+     */
+    | 'shown'
+
+    /**
+     * Is triggered at the end of the fade-out animation.
+     *
+     * Will pass the instance alongside the event.
+     *
+     * Won’t be trigger in a dataOnly configuration.
+     */
+    | 'hidden'
+
+    /**
+     * Is triggered at the end of destroy.
+     *
+     * Will pass the instance alongside the event.
+     */
+    | 'destroyed'
+
+    /**
+     * MBP’s Force Touch, iOS’s 3D Touch, Microsoft’s pressure or MDN’s force
+     *
+     *Is triggered when the pressure on the joystick is changed.
+     *
+     *The value, between 0 and 1, is sent back alongside the event.
+     */
+    | 'pressure';
+
+export type ManagerOnlyEventTypes =
+    /**
+     * A joystick just got added.
+     *
+     * Will pass the instance alongside the event.
+     */
+    | 'added'
+
+    /**
+     * A joystick just got removed.
+     *
+     * Fired at the end of the fade-out animation.
+     *
+     * Will pass the instance alongside the event.
+     *
+     * Won’t be trigger in a dataOnly configuration.
+     */
+    | 'removed';
+
+export type JoystickManagerEventTypes = JoystickEventTypes | ManagerOnlyEventTypes;
+
+export interface EventData {
+    type: JoystickEventTypes | ManagerOnlyEventTypes;
+    target: Collection;
+}
+
+export class JoystickManager {
+    static create(options?: JoystickManagerOptions): JoystickManager;
```

Comment:
Pretty sure this method is not actually static. The exported module treats it as such, but as it's a small module wrapper around a manager instance, that doesn't mean any manager instance can't call this as an instance method (not static class method.)

Reply:
looks like this technically returns a collection as well [manager.js:76](https://github.com/yoannmoinet/nipplejs/blob/master/src/manager.js#L76)

Reply:
Decided to just remove the static here. Manager and Collection overlap quite a bit, but the type def for collection would need to be fleshed out to make it work here and I don't have time. This is an adequate "1st cut", imo